ddtrace/settings/_inferred_service.py

```diff
@@ -0,0 +1,182 @@
+import os
+import fnmatch
+import pathlib
+
+INIT_PY = "__init__.py"
+ALL_PY_FILES = "*.py"
+GUNICORN_CMD_ARGS = "GUNICORN_CMD_ARGS"
+WSGI_APP_ENV = "WSGI_APP"
+
+
+class DetectionContext:
+    def __init__(self, envs):
+        self.envs = envs
+
+
+class ServiceMetadata:
+    def __init__(self, name):
+        self.name = name
+
+
+class PythonDetector:
+    def __init__(self, ctx):
+        self.ctx = ctx
+
+    def expected_command_name(self):
+        return 'python'
+
+    def detect(self, args):
+        prev_arg_is_flag = False
+        module_flag = False
+
+        for arg in args:
+            has_flag_prefix = arg.startswith("-")
+            is_env_variable = '=' in arg
+
+            should_skip_arg = prev_arg_is_flag or has_flag_prefix or is_env_variable
+
+            if module_flag:
+                return ServiceMetadata(arg), True
+
+            if not should_skip_arg:
+                wd = self.working_dir_from_envs(self.ctx.envs)
+                abs_path = self.abs_path(arg, wd)
+                if not pathlib.Path(abs_path).exists():
+                    return ServiceMetadata(""), False  # File not found
+                stripped = abs_path
+                if not pathlib.Path(stripped).is_dir():
+                    stripped = str(pathlib.Path(stripped).parent)
+                value, ok = self.deduce_package_name(stripped)
+                if ok:
+                    return ServiceMetadata(value), True
+                return ServiceMetadata(self.find_nearest_top_level(stripped)), True
+
+            if has_flag_prefix and arg == "-m":
+                module_flag = True
+
+            prev_arg_is_flag = has_flag_prefix
+
+        return ServiceMetadata(""), False
+
+    def working_dir_from_envs(self, envs):
+        # Assuming the function extracts working directory from envs
+        return os.getcwd()
+
+    def abs_path(self, a, wd):
+        # Converts the given path to an absolute path
+        return str(pathlib.Path(wd) / a)
+
+    # deduce_package_name is walking until a `__init__.py` is not found.
+    # All the dir traversed are joined then with `.` 
+    def deduce_package_name(self, fp):
+        up = str(pathlib.Path(fp).parent)
```

Comment:
I don't think we need to make these many conversions to `str` here given that we are reconverting them to `Path` anyway

Reply:
removed the str conversions